src/main/java/iudx/catalogue/server/mlayer/MlayerServiceImpl.java

```diff
@@ -259,19 +354,305 @@ public MlayerService getMlayerAllDatasets(
+      }
+    }
+    LOGGER.debug("getRGs for each provider type result succeeded");
+    int resourceGroupHits = 0;
+    JsonArray resourceGroup = new JsonArray();
+    LOGGER.debug("getRGs for each resource group result started");
+    for (int i = 0; i < size; i++) {
+      JsonObject record = datasets.getJsonArray(RESULTS).getJsonObject(i);
+      String itemType = Util.getItemType(record);
+      if (itemType.equals(ITEM_TYPE_RESOURCE_GROUP)) {
+        resourceGroupHits++;
+        record.put(
+            "providerDescription", providerDescription.getString(record.getString(PROVIDER)));
+        record.put("resourceServerRegURL", rsUrl.getString(record.getString(PROVIDER)));
+        record.put(
+            "cosURL", record.containsKey("cos") ? cosUrl.getString(record.getString("cos")) : "");
+
+        record.remove("cos");
+        resourceGroup.add(record);
+      }
+    }
+    LOGGER.debug("getRGs for each resource group result succeeded");
+    JsonObject resourceGroupResult =
+        new JsonObject()
+            .put("resourceGroupCount", resourceGroupHits)
+            .put("resourceGroup", resourceGroup);
+    LOGGER.debug("getRGs succeeded");
+    datasetResult.complete(resourceGroupResult);
+  }
+
+  private void allMlayerInstance(JsonObject instances, Promise<JsonObject> instanceResult) {
+    // Getting all instance names and icons
+    LOGGER.debug("getInstance started");
+    int instanceSize = instances.getJsonArray(RESULTS).size();
+    JsonObject instanceIcon = new JsonObject();
+    LOGGER.debug("getInstance for each instance started");
+    for (int i = 0; i < instanceSize; i++) {
+      JsonObject instanceObject = instances.getJsonArray(RESULTS).getJsonObject(i);
+      instanceIcon.put(
+          instanceObject.getString("name").toLowerCase(), instanceObject.getString("icon"));
+    }
+    LOGGER.debug("getInstance succeeded");
+    instanceResult.complete(instanceIcon);
+  }
+
+  private void gettingResourceAccessPolicyCount(
+      JsonObject resourceCountRes, Promise<JsonObject> resourceCountResult) {
+    // Getting resource item count
+    LOGGER.debug("resourceAP started");
+    JsonObject resourceItemCount = new JsonObject();
+    JsonObject resourceAccessPolicy = new JsonObject();
+    JsonArray resultsArray = resourceCountRes.getJsonArray(RESULTS);
+    LOGGER.debug("resourceAP for each resultsArray started");
+    resultsArray.forEach(
+        record -> {
+          JsonObject recordObject = (JsonObject) record;
+          String resourceGrp = recordObject.getString(KEY);
+          int docCount = recordObject.getInteger("doc_count");
+          resourceItemCount.put(resourceGrp, docCount);
+          Map<String, Integer> accessPolicy = new HashMap<>();
+          accessPolicy.put("PII", 0);
+          accessPolicy.put("SECURE", 0);
+          accessPolicy.put("OPEN", 0);
+
+          JsonArray accessPoliciesArray =
+              recordObject.getJsonObject("access_policies").getJsonArray("buckets");
```

Comment:
This line will give error when the elastic index will be empty. It can be handled using 204 error if output from elastic is empty with no results. Reference: https://github.com/datakaveri/iudx-catalogue-server/pull/364